pkg/kando/location.go

```diff
@@ -38,17 +47,86 @@ func newLocationCommand() *cobra.Command {
 	cmd.AddCommand(newLocationDeleteCommand())
 	cmd.PersistentFlags().StringP(pathFlagName, "s", "", "Specify a path suffix (optional)")
 	cmd.PersistentFlags().StringP(profileFlagName, "p", "", "Pass a Profile as a JSON string (required)")
-	_ = cmd.MarkFlagRequired(profileFlagName)
+	cmd.PersistentFlags().StringP(repositoryServerFlagName, "r", "", "Pass a Repository Server CR as a JSON string (required for kopia based blueprints)")
 	return cmd
 }
 
 func pathFlag(cmd *cobra.Command) string {
 	return cmd.Flag(pathFlagName).Value.String()
 }
 
+// validateCommandArgs makes sure that we are getting exactly
+// one of --profile or --repository-server flags
+func validateCommandArgs(cmd *cobra.Command) error {
+	profileFlag := cmd.Flags().Lookup(profileFlagName).Value.String()
+	repositoryServerFlag := cmd.Flags().Lookup(repositoryServerFlagName).Value.String()
+	if profileFlag != "" && repositoryServerFlag != "" {
+		return errors.New("Either --profile or --repository-server can be provided")
+	}
+	if profileFlag == "" && repositoryServerFlag == "" {
+		return errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+	}
+	return nil
+}
+
+func dataMoverFromCMD(cmd *cobra.Command, flag string) (datamover.DataMover, error) {
+	kopiaSnapshot, outputName := getKopiaSnapshotAndOutputNameFlag(cmd, flag)
+	dataMoverType, err := dataMoverTypeFromCMD(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	switch dataMoverType {
```

Comment:
```suggestion
	switch dataMoverTypeFromCMD(cmd) {
```
We are already having default case to return error. We don't need to return the same error in `dataMoverTypeFromCMD()`

Reply:
Done !